ynh-dev

```diff
@@ -1,382 +1,306 @@
+
+                warn "-------------------------------------------------------- "
+                warn "Launching gulp ...                                       "
+                warn "NB : This command will keep running and watch for changes"
+                warn " in the folder /ynh-dev/yunohost-admin/src, such that you"
+                warn "don't need to re-run npm yourself everytime you change   "
+                warn "something !                                              "
+                warn "-------------------------------------------------------- "
+                sudo su -c "gulp watch --dev" ynhdev
 
                 ;;
         esac
     done
+}
 
-
-elif [ "$1" = "test" ]; then
-    check_yunohost_vm
-    VERSION=$2
-
-    for i in ${!packages[@]}; do
-        case ${packages[i]} in
+function run_tests()
+{
+    assert_inside_vm
+    local PACKAGES="$@"
+    for PACKAGE in "$PACKAGES";
+    do
+        case $PACKAGE in
             yunohost)
                 # Pytest and tests dependencies
                 if ! type "pytest" > /dev/null; then
-                    echo "======================="
-                    echo "> Installing pytest ..."
-                    echo "======================="
-                    apt-get install python-pip
+                    info "> Installing pytest ..."
+                    apt-get install python-pip -y
                     pip2 install pytest
                 fi
-                PIP_DEPENDENCIES="pytest-mock requests-mock mock"
-                for DEP in $PIP_DEPENDENCIES
+                for DEP in "pytest-mock requests-mock mock"
                 do
                     if [ -z "$(pip show $DEP)" ]; then
-                        echo "======================="
-                        echo "Installing $DEP with pip"
-                        echo "======================="
+                        info "Installing $DEP with pip"
                         pip2 install $DEP
                     fi
                 done
 
                 # Apps for test
-                cd /vagrant/yunohost/src/yunohost/tests
-                if [ ! -d "apps" ]; then
-                    git clone https://github.com/YunoHost/test_apps ./apps
-                else
-                    cd apps
-                    git pull > /dev/null 2>&1
-                fi
+                cd /ynh-dev/yunohost/src/yunohost/tests
+                [ -d "apps" ] || git clone https://github.com/YunoHost/test_apps ./apps
+                cd apps
+                git pull > /dev/null 2>&1
 
                 # Run tests
                 echo "Running tests for YunoHost"
-                cd /vagrant/yunohost/
+                cd /ynh-dev/yunohost/
                 py.test tests
-                cd /vagrant/yunohost/src/yunohost
+                cd /ynh-dev/yunohost/src/yunohost
                 py.test tests
                 ;;
         esac
     done
+}
 
-
-elif [ "$1" = "self-update" ]; then
```

Comment:
Do you voluntarily remove this command ?

Reply:
Yes ... I don't feel like it's worth creating a command for it. It just does a "git pull" (and inside the vm, which is weird ?)